lua/ibl/init.lua

```diff
@@ -260,6 +265,49 @@ M.refresh = function(bufnr)
         end
     end
 
+    if scope and scope_start_line then
+        -- find whitespace tables for the start and end lines of scope
+        local whitespace_start = utils.get_whitespace(scope_start_line[1])
+        local whitespace_end = utils.get_whitespace(scope_end_line[1])
+        local whitespace_tbl_start = indent.get(whitespace_start, indent_opts, nil)
+        local whitespace_tbl_end = indent.get(whitespace_end, indent_opts, nil)
+        local whitespace, whitespace_tbl
+        -- use the smallest whitespace table of the two to determine the scope index
+        if #whitespace_tbl_end < #whitespace_tbl_start then
+            whitespace = whitespace_end
+            whitespace_tbl = whitespace_tbl_end
+        else
+            whitespace = whitespace_start
+            whitespace_tbl = whitespace_tbl_start
+        end
+
+        -- do the same calculations as in the main loop below, but note that the scope
+        -- start and end will never be on a blankline, so these cases simplify a lot
+        local current_left_offset = left_offset
+        while #whitespace_tbl > 0 and current_left_offset > 0 do
+            table.remove(whitespace_tbl, 1)
+            current_left_offset = current_left_offset - 1
+        end
```

Comment:
The horizontal scroll fix is exactly the same as in the main loop, I think it would be nicer to make this a util function.

Reply:
I made a util function now. Please let me know if you want me to change anything.